llvm/lib/IR/Function.cpp

```diff
@@ -962,13 +962,29 @@ static std::string getMangledTypeStr(Type *Ty, bool &HasUnnamedType) {
     case Type::IntegerTyID:
       Result += "i" + utostr(cast<IntegerType>(Ty)->getBitWidth());
       break;
-    case Type::GaloisFieldTyID:
-      Result += "field";
-      break;
-    case Type::EllipticCurveTyID:
-      Result += "curve";
+    case Type::GaloisFieldTyID: {
+      switch (cast<GaloisFieldType>(Ty)->getFieldKind()) {
+        case GaloisFieldKind::GALOIS_FIELD_BLS12381_BASE:     Result += "field_bls12381_base";     break;
```

Comment:
Please use `GaloisFieldTypes.def` file to cover all the enum elements and make it easier to add new types. [Example](https://github.com/NilFoundation/zkllvm-circifier/blob/master/clang/lib/CodeGen/CodeGenTypes.cpp#L336).